ui/src/app/pages/login/account-creation/account-creation-survey.tsx

```diff
@@ -0,0 +1,202 @@
+import {CheckBox, RadioButton} from 'app/components/inputs';
+import colors from 'app/styles/colors';
+import {
+  DataAccessLevel,
+  Gender,
+  Profile,
+  Race
+} from 'generated/fetch';
+import {Section} from './account-creation';
+
+import {
+  profileApi
+} from 'app/services/swagger-fetch-clients';
+import {toggleIncludes} from 'app/utils';
+import {AccountCreationOptions} from './account-creation-options';
+
+
+const styles = {
+  questionLabel: {
+    width: 76,
+    color: colors.primary,
+    fontFamily: 'Montserrat',
+    fontSize: '14px',
+    fontWeight: 400,
+  }
+};
+
+export const DropDownSection = (props) => {
+  return <Section header={props.header}>
+    <Dropdown placeholder='Select' options={props.options} style={{width: '50%'}}
+              value={props.value}
+              onChange={(e) => props.onChange(e.value)}/>
+  </Section>;
+};
+
+export const CheckBoxWithLabel = (props) => {
+  return <div style={{display: 'flex', width: '9rem', marginBottom: '0.5rem'}}>
+    <CheckBox style={{marginTop: '0.3rem'}}
+              checked={props.checked} onChange={(value) => props.onChange(value)}/>
+    <div style={{paddingLeft: '0.5rem', paddingRight: '0.5rem'}}>
+      <label style={{...styles.questionLabel, ...props.style}}>{props.label}</label>
+    </div>
+  </div>;
+};
+
+export interface AccountCreationSurveyProps {
+  invitationKey: string;
+  profile: Profile;
+  setProfile: Function;
+}
+
+export interface AccountCreationState {
+  creatingAccount: boolean;
+  profile: Profile;
+}
+
+export class AccountCreationSurvey extends React.Component<AccountCreationSurveyProps, AccountCreationState> {
+  constructor(props: any) {
+    super(props);
+    this.state = {
+      creatingAccount: false,
+      profile: {
+        username: '',
+        dataAccessLevel: DataAccessLevel.Protected,
+        demographicSurvey: {
+          race: [] as Race[],
+          ethnicity: undefined,
+          gender: [] as Gender[],
+          yearOfBirth: 0,
+          education: undefined,
+          disability: false
+        }
+      }
+    };
+  }
+
+  componentDidMount() {
+    this.setState({profile: this.profileObj});
+  }
+
+  get profileObj() {
+    const demographicSurvey = this.state.profile.demographicSurvey;
+    return {
+      ...this.props.profile,
+      demographicSurvey
+    };
+  }
+
+  createAccount(): void {
+    const {invitationKey, setProfile} = this.props;
+    this.setState({creatingAccount: true});
+    profileApi().createAccount({profile: this.profileObj, invitationKey: invitationKey})
+      .then((savedProfile) => {
+        this.setState({profile: savedProfile, creatingAccount: false});
+        setProfile(savedProfile, 'accountCreationSuccess');
+      }).catch(error => {
+        console.log(error);
+        this.setState({creatingAccount: false});
+      });
+  }
+
+  updateGenderList(attribute, value) {
+    const genderList = toggleIncludes(value, this.state.profile.demographicSurvey[attribute]);
```

Comment:
What does this do?

Reply:
It removes the gender_value from the array if it exist and adds if not

Reply:
ok, thanks!

Could you change the name of this const as well to reflect the new function name, and also add a comment above the function describing what it does?